packages/plugin-audio/web/src/toolbar/modals/AudioSettings.tsx

```diff
@@ -0,0 +1,176 @@
+  SettingsPanelHeader,
+  SettingsMobileHeader,
+  SettingsSeparator,
+  SettingsAddItem,
+  InputWithLabel,
+  Label,
+} from 'wix-rich-content-ui-components';
+
+const AudioSettings = ({
+  onCancel,
+  updateData,
+  componentData,
+  helpers,
+  pubsub,
+  theme,
+  t,
+  isMobile,
+  settings,
+  accept,
+}) => {
+  const [disableDownload, setDisableDownload] = useState(
+    componentData.disableDownload ?? !!settings.disableDownload
+  );
+  const [audioName, setAudioName] = useState(componentData?.name || '');
+  const [authorName, setAuthorName] = useState(componentData?.authorName || '');
+  const [coverImage, setCoverImage] = useState(componentData?.coverImage || null);
+  const styles = mergeStyles({ styles: Styles, theme });
+
+  const onSave = () => {
+    const newComponentData = {
+      ...componentData,
+      disableDownload,
+      coverImage,
+      authorName,
+      name: audioName,
+    };
+    updateData(newComponentData);
+    helpers?.closeModal();
+  };
+
+  const onDownlandToggle = () => setDisableDownload(!disableDownload);
+
+  const handleFilesAdded = args => {
+    const coverImage = args.data[0] || args.data;
+    setCoverImage(coverImage);
+  };
+
+  const handleFileSelection = () => {
+    const deleteBlock = pubsub.get('deleteBlock');
+    helpers.handleFileSelection(undefined, false, handleFilesAdded, deleteBlock, componentData);
+  };
+
+  const handleFileChange = file => {
+    helpers.handleFileUpload(file, handleFilesAdded);
+  };
+
+  const handleCoverImageDelete = () => setCoverImage(null);
+
+  const inputsData = [
+    {
+      label: t('AudioPlugin_Settings_AudioName_Label'),
+      value: audioName,
+      onChange: caption => setAudioName(caption),
+      dataHook: 'audioSettingsAudioNameInput',
+    },
+    {
+      label: t('AudioPlugin_Settings_AuthorName_Label'),
+      value: authorName,
+      onChange: caption => setAuthorName(caption),
+      dataHook: 'audioSettingsAuthorNameInput',
+    },
+  ];
+
+  const renderInputs = () =>
+    inputsData.map(props => <InputWithLabel key={props.dataHook} theme={theme} {...props} />);
+
+  const renderHeader = () =>
+    isMobile ? (
+      <SettingsMobileHeader
+        t={t}
+        theme={theme}
+        title={t('AudioPlugin_Settings_Header')}
+        onCancel={onCancel}
+        onSave={onSave}
+        useNewSettingsUi
+      />
+    ) : (
+      <SettingsPanelHeader title={t('AudioPlugin_Settings_Header')} onClose={onCancel} />
+    );
+
+  const imgSrc =
+    coverImage &&
+    getImageSrc(coverImage, helpers?.getImageUrl, {
+      requiredWidth: 120,
+      requiredHeight: 120,
+      requiredQuality: 70,
+    });
+
+  useEffect(() => {
+    const newComponentData = {
+      ...componentData,
+      authorName,
+      name: audioName,
+      disableDownload,
+      coverImage,
+    };
+    updateData(newComponentData);
+  }, [audioName, authorName, disableDownload, coverImage]);
+
+  return (
+    <div
+      data-hook="settings"
```

Comment:
```suggestion
      data-hook="audio-settings"
```

or remove if not necessary